File: 剑指offer/链表/18.删除链表的节点/18.删除链表的节点.go
package 链表

import "algorithms-code-note/剑指offer/structures"

type ListNode = structures.ListNode

func deleteNode(head *ListNode, val int) *ListNode {
	dummy := &ListNode{0, head}
	pre := dummy
	cur := head
	for cur != nil {
		if cur.Val == val {
			pre.Next = cur.Next
		}
		pre = pre.Next
		cur = cur.Next
	}
	return dummy.Next
}

File: 剑指offer/链表/18.删除链表的节点/18.删除链表的节点.py
from util.leetcode_type import ListNode

"""
dummy作为head前节点 方便返回内容
pre作为cur的前节点，方便删除cur的内容
"""

class Solution:
    def deleteNode(self, head: ListNode, val: int) -> ListNode:
        dummy = ListNode(0, head)
        pre, cur = dummy, head
        while cur:
            if cur.val == val:
                pre.next = cur.next
            cur = cur.next
            pre = pre.next
        return dummy.next
